File: proj.android_studio/ee-x-ads/src/main/java/com/ee/ads/InterstitialAdHelper.java
package com.ee.ads;

import androidx.annotation.NonNull;

import com.ee.core.IMessageBridge;
import com.ee.core.internal.Utils;

/**
 * Created by Zinge on 10/13/17.
 */

public class InterstitialAdHelper {
    private final IMessageBridge _bridge;
    private final IInterstitialAd _ad;
    private final MessageHelper _helper;

    public InterstitialAdHelper(@NonNull IMessageBridge bridge,
                                @NonNull IInterstitialAd ad,
                                @NonNull MessageHelper helper) {
        _bridge = bridge;
        _ad = ad;
        _helper = helper;
    }

    public void registerHandlers() {
        _bridge.registerHandler(_helper.isLoaded(), message ->
            Utils.toString(_ad.isLoaded()));

        _bridge.registerHandler(_helper.load(), message -> {
            _ad.load();
            return "";
        });

        _bridge.registerHandler(_helper.show(), message -> {
            _ad.show();
            return "";
        });
    }

    public void deregisterHandlers() {
        _bridge.deregisterHandler(_helper.isLoaded());
        _bridge.deregisterHandler(_helper.load());
        _bridge.deregisterHandler(_helper.show());
    }
}

File: proj.android_studio/ee-x-ads/src/main/java/com/ee/ads/InterstitialAdHelper.kt
package com.ee.ads

import androidx.annotation.AnyThread
import com.ee.core.IMessageBridge
import com.ee.core.internal.Utils
import com.ee.core.registerHandler

/**
 * Created by Zinge on 10/13/17.
 */
class InterstitialAdHelper(private val _bridge: IMessageBridge,
                           private val _ad: IInterstitialAd,
                           private val _helper: MessageHelper) {
    @AnyThread
    fun registerHandlers() {
        _bridge.registerHandler(_helper.isLoaded) {
            Utils.toString(_ad.isLoaded)
        }
        _bridge.registerHandler(_helper.load) {
            _ad.load()
            ""
        }
        _bridge.registerHandler(_helper.show) {
            _ad.show()
            ""
        }
    }

    @AnyThread
    fun deregisterHandlers() {
        _bridge.deregisterHandler(_helper.isLoaded)
        _bridge.deregisterHandler(_helper.load)
        _bridge.deregisterHandler(_helper.show)
    }
}